wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceRuntimes.java

```diff
@@ -272,23 +256,74 @@ public WorkspaceStatus getStatus(String workspaceId) {
     }
   }
 
+  private class StartRuntimeTask implements Runnable {
+    private final Workspace workspace;
+    private final Map<String, String> options;
+    private final String workspaceId;
+    private final InternalRuntime runtime;
+
+    public StartRuntimeTask(
+        Workspace workspace, Map<String, String> options, InternalRuntime runtime) {
+      this.workspace = workspace;
+      this.options = options;
+      this.workspaceId = workspace.getId();
+      this.runtime = runtime;
+    }
+
+    @Override
+    public void run() {
+      try {
+        runtime.start(options);
+        runtimes.replace(workspaceId, new RuntimeState(runtime, RUNNING));
+
+        LOG.info(
+            "Workspace '{}:{}' with id '{}' started by user '{}'",
+            workspace.getNamespace(),
+            workspace.getConfig().getName(),
+            workspace.getId(),
+            sessionUserNameOr("undefined"));
+        publishWorkspaceStatusEvent(workspaceId, RUNNING, STARTING, null);
+      } catch (InfrastructureException e) {
+        runtimes.remove(workspaceId);
+        String failureCause = "failed";
+        if (e instanceof RuntimeStartInterruptedException) {
+          failureCause = "interrupted";
+          publishWorkspaceStatusEvent(workspaceId, STOPPED, STARTING, e.getMessage());
+        }
+        LOG.info(
+            "Workspace '{}:{}' with id '{}' start {}",
+            workspace.getNamespace(),
+            workspace.getConfig().getName(),
+            workspace.getId(),
```

Comment:
Should not `workspaceId` variable be here? 
And are you sure that workspaceId should be `field` but not a local variable? It's OK for and it is just a question =)

Reply:
I didn't see much difference but will move to local var